core/src/replay_stage.rs

```diff
@@ -3186,11 +3193,17 @@ impl ReplayStage {
             num_active_banks,
             active_bank_slots
         );
-        if num_active_banks > 0 {
-            let replay_result_vec = if num_active_banks > 1 && replay_slots_concurrently {
+        if active_bank_slots.is_empty() {
+            return false;
+        }
```

Comment:
I thought that this was a simple enough sanity check to pull out and explicitly return early for - this enabled us to remove a level of indentation for everything else

Reply:
Never nesters unite

Reply:
early returns make code more complex and easier to introduce bugs in the future. CHANGE MY MIND

Reply:
Number of bugs in a function is proportional to the maximum indent. CHANGE MY MIND

Reply:
🍿

Reply:
> early returns make code more complex and easier to introduce bugs in the future

https://github.com/steviez/solana/commit/4bcbbbe825f38f998f54f22329a75c9d7ddfc2ce?diff=unified&w=0

Reply:
> Number of bugs in a function is proportional to the maximum indent. CHANGE MY MIND

you're stuck in the present old man. join us in the future!